lightning/src/onion_message/messenger.rs

```diff
@@ -243,6 +244,89 @@ impl OnionMessageRecipient {
 	}
 }
 
+pub trait RespondFunction<T: OnionMessageContents>{
+	fn respond(&self, response: T, reply_path: BlindedPath);
+}
+
+impl<F, T> RespondFunction<T> for F
+where
+T: OnionMessageContents,
+F: Fn(T, BlindedPath)
+{
+	fn respond(&self, response: T, reply_path: BlindedPath) {
+		self(response, reply_path)
+	}
+}
+
+/// A struct handling response to an [`OnionMessage`]
+pub struct Responder<R, T>
+where
+R: RespondFunction<T>,
+T: OnionMessageContents
+{
+	messenger_function: R,
+	pub reply_path: BlindedPath,
```

Comment:
Does this need to be `pub`?

Reply:
I think this was added for the previous version of the PR.

Addressed in [2806122](https://github.com/lightningdevkit/rust-lightning/pull/2907/commits/2806122f65d57d3a6a91591aff6e9ccb7590af13)
Thanks!